internal/controllers/clusterclass/clusterclass_controller.go

```diff
@@ -244,14 +251,18 @@ func (r *Reconciler) reconcileVariables(ctx context.Context, clusterClass *clust
 						continue
 					}
 					uniqueVars[variable.Name] = true
-					clusterClass.Status.Variables = append(clusterClass.Status.Variables, statusVariableFromClusterClassVariable(variable, patch.Name))
+					ccStatusVars = append(ccStatusVars, statusVariableFromClusterClassVariable(variable, patch.Name))
 				}
 			}
 		}
 	}
 	if len(errs) > 0 {
+		conditions.MarkFalse(clusterClass, clusterv1.ClusterClassVariablesReconciledCondition, clusterv1.VariableDiscoveryFailedReason, clusterv1.ConditionSeverityError,
+			"VariableDiscovery failed: %s", kerrors.NewAggregate(errs))
 		return errors.Wrapf(kerrors.NewAggregate(errs), "failed to discover variables for ClusterClass %s", clusterClass.Name)
 	}
+	clusterClass.Status.Variables = ccStatusVars
```

Comment:
Q: This behavior is consistent with what we do for ExtensionConfig discovery right? (if success => update the status.{variables,handlers}, if failure => set condition to false and don't update the status.{variables,handlers})

Reply:
Yeah - it's the same. I've added a TODO about whether or not to just blank out the variables if we can't reconcile them. Not sure what's the best behaviour here from a user perspective, but I think we can come back to it.

Reply:
I think it's fine to have the same behavior